### DISCUSSION TITLE
Prometheus kube metrics

### DISCUSSION THREAD

**Discussion by Angi2412** [linkerd/linkerd2#5502] [Q&A]
Hello,

I set up linkerd including Prometheus and found out that Kubernetes metrics such as e.g. "kube_pod_container_resource_limits_cpu_cores" are not available. 
Is it possible to make them available or do I have to set up my own Prometheus?

**Comment by olix0r** [comment]
@Angi2412 The Prometheus instance shipped with Linkerd includes only the metrics needed to power the Linkerd CLI and Web interface. We recommend running a separate Prometheus instance if you want to include additional metrics or deviate from the default configuration. Note that Linkerd can be configured to use this alternate instance so that you can disable the instance in the Linkerd namespace.

**Comment by Angi2412** [comment]
@olix0r Thank you for the fast reply. 
1. How does the linkerd job Scarpe description (content of file.yaml) look exactly?
I want to use it via the `helm install -f file.yaml` command.

2. What about the `edge-21.1.1` release? 
I don't know if I understood it correctly, but isn't there an option to modify the Prometheus installation?
The [release notes](https://github.com/linkerd/linkerd2/releases/tag/edge-21.1.1) say:
"[...] Additionally, a **new linkerd-viz extension** has been introduced that separates
the installation of the Grafana, **Prometheus**, web, and tap components. [...]
The linkerd viz install command does not have any cli flags to customize the
install directly, but instead **follows the Helm way of customization by using
flags such as set, set-string, values, set-files**."

**Reply by kleimkuhler** [reply]
@Angi2412 if I can try to clarify what Oliver was saying, we recommend in this case that you use your own external Prometheus deployment and configure to it scrape the control plane as well as the proxy's metrics.

You'll find the scrape config file as well as how to modify a Linkerd installation to point to an external Prometheus installation in the [Bring Your Own Prometheus](https://linkerd.io/2/tasks/external-prometheus/) section of our docs.

Let me know if that helps!

**Reply by Angi2412** [reply]
@kleimkuhler ok, thank you for clarifying. I will try the recommendation.
